### ISSUE TITLE
Sidebar with TOC accessibility

### ISSUE THREAD

**Issue by viktor-yakubiv** [oacore/reader#14]
It would be nice to have proper markup for table of contents in the sidebar.

### Lists

In spite of TOC is a table, in general contents is a list of items with links and inner lists with the same pattern. To be more specific, in this case the order is important, so the list should be ordered, i.e. `<ol>`. The pattern is similar to a navigation.

### Buttons

List expanding should be accessible for keyboard users. The action does not transit the page so the proper element is a _button_.

### State reporting

When menus expand/collapse, the state should be actually represented to DOM with `aria-*` attributes. In this case, we need `role=treeitem` and `aria-expanded=true/false`.

---

Basically, the structure should be the following:

```html
<ol>
  <li role=treeitem aria-expanded=false>
    <a href="#1">Section 1 title</a>
    <button>Expand Section 1 menu</button>
    <ol>
      <li>
        <a href="#1.1">Subsection 1.1 title</a>
      <li>
        <a href="#1.2">Subsection 1.2 title</a>
    </ol>
  <li class=open role=treeitem aria-expanded=true>
    <a href="#2">Section 2 title</a>
    <button>Collapse Section 2 menu</button>
    <ol>
      <li>
        <a href="#2.1">Subsection 2.1 title</a>
      <li>
        <a href="#2.2">Subsection 2.2 title</a>
      <li>
        <a href="#2.3">Subsection 2.3 title</a>
    </ol>
  <li>
    <a href="#3">Section 3 title</a>
</ol>
```

Note: [tag omission](https://html.spec.whatwg.org/multipage/syntax.html#optional-tags) is not allowed in JSX.
